scripts/linters/pylint_extensions.py

```diff
@@ -1854,6 +1820,46 @@ def process_tokens(self, tokens):
                         'single-line-pragma', line=line_num)
 
 
+class SingleSpaceAfterKeyWordChecker(checkers.BaseChecker):
+    """Custom pylint checker which checks that there is a single space
+    between a keyword and a bracket `(` if a bracket is used. Keywords
+    tested are `if`, `elif`, `while`, and `yield`.
+    """
+
+    __implements__ = interfaces.ITokenChecker
+
+    name = 'single-space-after-keyword'
+    priority = -1
+    msgs = {
+        'C0029': (
+            'Please add a single space after `%s` statement.',
+            'single-space-after-keyword',
+            'A single space should be added after a keyword.',
+        ),
+    }
+
+    def process_tokens(self, tokens):
+        """Custom pylint checker which makes sure that every `if`, `elif`,
+        `while`, and `yield` statement is followed by a single space.
+
+        Args:
+            tokens: Token. Object to access all tokens of a module.
+        """
+        keywords = set(['if', 'elif', 'while', 'yield'])
+        for (token_type, token, (line_num, _), _, line) in tokens:
+            if token_type == tokenize.NAME and token in keywords:
+                line = line.strip()
+                if re.search(br'^' + token + br'\s*\(', line):
```

Comment:
Good question -- I think we would want to keep the parenthesis, since we wouldn't want the checker to flag something like `ifxxx = "hello!"` as being an error since we need a space after `if`.

Reply:
Oh, I see what you mean. You're right!